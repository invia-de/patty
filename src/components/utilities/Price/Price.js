import React from 'react';
import PropTypes from 'prop-types';

/**
 * formats a number into german number/currency format with thousand delimiters,
 * currency symbol (only CHF and EUR) and fraction units
 *
 * @author [Eric Zieger](mailto:eric.zieger@invia.de)
 */
export default function Price({ value, decimals, symbol, currency }) {
  const fractionCount = decimals ? 2 : 0;

  value = parseFloat(value).toLocaleString('de-DE', {
    minimumFractionDigits: fractionCount,
    maximumFractionDigits: fractionCount
  });

  if (symbol === 'none') {
    return value;
  }

  currency = currency.toUpperCase() === 'CHF' ? 'CHF' : 'â‚¬';

  return (
    <span>
      {symbol === 'before' ? currency + ' ' + value : value + ' ' + currency}
    </span>
  );
}
Price.propTypes = {
  /** the number you want to format into a price */
  value: PropTypes.number.isRequired,
  /** if you want fraction counts or not */
  decimals: PropTypes.bool,
  /** currency symbol position */
  symbol: PropTypes.oneOf(['before', 'after', 'none']),
  /** currency symbol to use either `EUR` or `CHF` */
  currency: PropTypes.string
};
Price.defaultProps = {
  currency: 'EUR',
  symbol: 'before'
};
