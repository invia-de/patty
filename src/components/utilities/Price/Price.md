```jsx
<Price value={123} decimals symbol="before"/>
<br />
<Price value={1546000} decimals symbol="none"/>
<br />
<Price value={444} decimals symbol="after"/>
```
