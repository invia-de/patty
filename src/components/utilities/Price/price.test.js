import React from 'react';
import Price from './Price';
import { render } from 'react-testing-library';

// https://stackoverflow.com/questions/26124914/how-to-test-react-proptypes-through-jest

test('Price renders correctly (default)', () => {
  const { container } = render(<Price value={90123.01} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Price renders correctly (with decimals)', () => {
  const { container } = render(<Price value={90123.01} decimals />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Price renders correctly (with decimals, symbol at the end)', () => {
  const { container } = render(
    <Price value={90123.01} decimals symbol="after" />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('Price renders correctly (with decimals, no symbol)', () => {
  const { container } = render(
    <Price value={90123.01} decimals symbol="none" />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('Price renders correctly (CHF symbol)', () => {
  const { container } = render(<Price value={90123.01} currency="CHF" />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Price renders (decimals, from `integer`)', () => {
  const { container } = render(<Price value={1282} decimals />);
  expect(container.firstChild).toMatchSnapshot();
});
