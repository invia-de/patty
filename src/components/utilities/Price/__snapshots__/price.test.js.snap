// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders (decimals, from \`integer\`) 1`] = `
<span>
  1,282.00 €
</span>
`;

exports[`Price renders correctly (CHF symbol) 1`] = `
<span>
  90,123 CHF
</span>
`;

exports[`Price renders correctly (default) 1`] = `
<span>
  90,123 €
</span>
`;

exports[`Price renders correctly (with decimals) 1`] = `
<span>
  90,123.01 €
</span>
`;

exports[`Price renders correctly (with decimals, no symbol) 1`] = `
<span>
  90,123.01 €
</span>
`;

exports[`Price renders correctly (with decimals, symbol at the end) 1`] = `
<span>
  90,123.01 €
</span>
`;
